src/components/auth/SetupForm.tsx

```diff
@@ -0,0 +1,125 @@
+import { useTranslation } from 'react-i18next';
+import { Button, Checkbox, Dropdown, Form, Input } from 'semantic-ui-react';
+import React, { ChangeEvent, useEffect, useRef, useState } from 'react';
+import { Gender } from '../../clients/server.generated';
+import { Dispatch } from 'redux';
+import { connect } from 'react-redux';
+import { authSetup } from '../../stores/auth/actionCreators';
+import { useNavigate } from 'react-router-dom';
+import { NavigateFunction } from 'react-router/dist/lib/hooks';
+
+interface Props {
+  setup: (
+    email: string,
+    firstname: string,
+    lastname: string,
+    gender: string,
+    password: string,
+    rememberMe: boolean,
+    navigate: NavigateFunction) => void;
+}
```

Comment:
Please run the linter here as well, because I do not think ESlint will allow this (`npm run lint:fix`)